// @testing-library/jest-native/dist/extend-expect.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const to_be_disabled_1 = require("./to-be-disabled");
const to_be_empty_element_1 = require("./to-be-empty-element");
const to_contain_element_1 = require("./to-contain-element");
const to_have_prop_1 = require("./to-have-prop");
const to_have_style_1 = require("./to-have-style");
const to_have_text_content_1 = require("./to-have-text-content");
const to_be_visible_1 = require("./to-be-visible");
const to_have_accessibility_state_1 = require("./to-have-accessibility-state");
const to_have_accessibility_value_1 = require("./to-have-accessibility-value");
expect.extend({
    toBeDisabled: to_be_disabled_1.toBeDisabled,
    toBeEnabled: to_be_disabled_1.toBeEnabled,
    toBeEmptyElement: to_be_empty_element_1.toBeEmptyElement,
    toBeEmpty: to_be_empty_element_1.toBeEmpty,
    toContainElement: // Deprecated
    to_contain_element_1.toContainElement,
    toHaveProp: to_have_prop_1.toHaveProp,
    toHaveStyle: to_have_style_1.toHaveStyle,
    toHaveTextContent: to_have_text_content_1.toHaveTextContent,
    toBeVisible: to_be_visible_1.toBeVisible,
    toHaveAccessibilityState: to_have_accessibility_state_1.toHaveAccessibilityState,
    toHaveAccessibilityValue: to_have_accessibility_value_1.toHaveAccessibilityValue,
});
