// @testing-library/react-native/build/queries/role.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.bindByRoleQueries = void 0;
var _accessiblity = require("../helpers/accessiblity");
var _accessibilityState = require("../helpers/matchers/accessibilityState");
var _matchStringProp = require("../helpers/matchers/matchStringProp");
var _within = require("../within");
var _makeQueries = require("./makeQueries");
const matchAccessibleNameIfNeeded = (node, name) => {
  if (name == null) return true;
  const {
    queryAllByText,
    queryAllByLabelText
  } = (0, _within.getQueriesForElement)(node);
  return queryAllByText(name).length > 0 || queryAllByLabelText(name).length > 0;
};
const matchAccessibleStateIfNeeded = (node, options) => {
  return options != null ? (0, _accessibilityState.matchAccessibilityState)(node, options) : true;
};
const queryAllByRole = instance => function queryAllByRoleFn(role, options) {
  return instance.findAll(node =>
  // run the cheapest checks first, and early exit too avoid unneeded computations

  typeof node.type === 'string' && (0, _matchStringProp.matchStringProp)(node.props.accessibilityRole, role) && matchAccessibleStateIfNeeded(node, options) && matchAccessibleNameIfNeeded(node, options?.name));
};
const buildErrorMessage = (role, options = {}) => {
  const errors = [`role: "${String(role)}"`];
  if (options.name) {
    errors.push(`name: "${String(options.name)}"`);
  }
  _accessiblity.accessibilityStateKeys.forEach(stateKey => {
    if (options[stateKey] !== undefined) {
      errors.push(`${stateKey} state: ${options[stateKey]}`);
    }
  });
  return errors.join(', ');
};
const getMultipleError = (role, options) => `Found multiple elements with ${buildErrorMessage(role, options)}`;
const getMissingError = (role, options) => `Unable to find an element with ${buildErrorMessage(role, options)}`;
const {
  getBy,
  getAllBy,
  queryBy,
  queryAllBy,
  findBy,
  findAllBy
} = (0, _makeQueries.makeQueries)(queryAllByRole, getMissingError, getMultipleError);
const bindByRoleQueries = instance => ({
  getByRole: getBy(instance),
  getAllByRole: getAllBy(instance),
  queryByRole: queryBy(instance),
  queryAllByRole: queryAllBy(instance),
  findByRole: findBy(instance),
  findAllByRole: findAllBy(instance)
});
exports.bindByRoleQueries = bindByRoleQueries;
